scripts/generate-framework-agreement-counterpart-signature-pages.py

```diff
@@ -131,7 +136,7 @@ def approve_supplier_framework_agreement(record):
                         record["countersignedAt"] = countersigned_agreement["countersignedAgreementReturnedAt"]
                     except dmapiclient.errors.HTTPError as e:
                         logger.warn(f"failed to countersign agreement {agreement_id} for supplier {supplier_id}: {e}")
-
+                unsuspend_suspended_supplier_services(record, AUTOMATED_SUSPENDING_USER, client, logger, dry_run)
```

Comment:
What happens if countersigning has failed for this supplier?

Reply:
Good question I guess if approving fails we should also not unsuspend, as this will leave the supplier in a cleaner state for the subsequent run. In that case I can move them inside the try for approval.